$OpenBSD: patch-vncserver,v 1.1.1 2022/01/24 01:12:58 unixman Exp $

Patch away check which doesn't work on OpenBSD.

Fix by unixman: comment out the following line. It does not work on OpenBSD 7.0 !
$default_opts{rfbwait} = 30000;

Index: vncserver
--- vncserver.orig
+++ vncserver
@@ -205,7 +205,7 @@
 $default_opts{geometry} = $geometry if ($geometry);
 $default_opts{depth} = $depth if ($depth);
 $default_opts{pixelformat} = $pixelformat if ($pixelformat);
-$default_opts{rfbwait} = 30000;
+#$default_opts{rfbwait} = 30000;
 $default_opts{rfbauth} = "$vncUserDir/passwd";
 $default_opts{rfbport} = $vncPort;
 $default_opts{fp} = $fontPath if ($fontPath);
@@ -523,22 +523,6 @@
 sub CheckDisplayNumber
 {
     local ($n) = @_;
-
-    socket(S, $AF_INET, $SOCK_STREAM, 0) || die "$prog: socket failed: $!\n";
-    eval 'setsockopt(S, &SOL_SOCKET, &SO_REUSEADDR, pack("l", 1))';
-    if (!bind(S, pack('S n x12', $AF_INET, 6000 + $n))) {
-	close(S);
-	return 0;
-    }
-    close(S);
-
-    socket(S, $AF_INET, $SOCK_STREAM, 0) || die "$prog: socket failed: $!\n";
-    eval 'setsockopt(S, &SOL_SOCKET, &SO_REUSEADDR, pack("l", 1))';
-    if (!bind(S, pack('S n x12', $AF_INET, 5900 + $n))) {
-	close(S);
-	return 0;
-    }
-    close(S);
 
     if (-e "/tmp/.X$n-lock") {
 	warn "\nWarning: $host:$n is taken because of /tmp/.X$n-lock\n";
